mclf/berkovich/berkovich_line.py

```diff
@@ -1278,25 +1275,23 @@ def normalized_reduction(v, f):
     a nonzero element fb in the residue field of v ???
     Precise definition needed!
     """
-
     F = v.domain()
-    R = F._ring
     x = F.gen()
     r = v(f)
     m = abs(r.denominator())
     v1 = v._base_valuation
     if hasattr(v1, 'equivalence_unit'):
         fl = v.reduce(f^m*F(v1.equivalence_unit(-m*r))).factor()
         if len(fl)>0:
-            return fl[0][0]^sign(fl[0][1])
+            return fl[0][0]^sgn(fl[0][1])
```

Comment:
There are a lot of `^` in this area that should probably be `**`?

Reply:
Yes! For a long time I was completely ignorant of the ^-issue in python, and I still do this wrong from time to time.